src/transformers/integrations.py

```diff
@@ -0,0 +1,49 @@
+## Integrations with other Python libraries
+
+import os
+
+
+try:
+    import comet_ml
+
+    _has_comet = True
+except (ImportError):
+    _has_comet = False
+
+
+try:
+    import wandb
+
+    wandb.ensure_configured()
+    if wandb.api.api_key is None:
+        _has_wandb = False
+        wandb.termwarn("W&B installed but not logged in.  Run `wandb login` or set the WANDB_API_KEY env variable.")
+    else:
+        _has_wandb = False if os.getenv("WANDB_DISABLED") else True
+except (ImportError, AttributeError):
+    _has_wandb = False
+
+
+try:
```

Comment:
The corresponding lines could be removed from `trainer.py` now.

Reply:
Good, done.